torchdet3d/utils/tracking_tools.py

```diff
@@ -0,0 +1,308 @@
+            # then we smooth kps with EMA
+            if add_dist < add_treshold:
+                self.no_updated_frames = 0
+                filtered_kps = (1 - filter_speed) * self.kps[-2] + filter_speed * considered_kps
+            elif self.no_updated_frames > no_updated_frames_treshold:
+                # if bbox haven't been updated too long -> interrupt EMA
+                # and get new bbox
+                filtered_kps = considered_kps
+            else:
+                # if not -> use bbox from previous frame
+                filtered_kps = self.kps[-2]
+                self.no_updated_frames += 1
+
+            self.kps[-1] = tuple(filtered_kps.reshape(-1).tolist())
+
+    def _align_kp_positions(self):
+        # store indexes for matching
+        indexes = dict(zip(range(9), range(9)))
+        # list for marking vertexes
+        ind_updated = [False for i in range(9)]
+        for i in range(len(self.kps[-1])):
+            if ind_updated[i]:
+                continue
+            distance = np.linalg.norm(self.kps[-1][i, :] - self.kps[-2][i, :])
+            min_d_idx = i
+            for j in range(i+1, len(self.kps[-1])):
+                d = np.linalg.norm(self.kps[-1][i, :] - self.kps[-2][j, :])
+                if d < distance:
+                    min_d_idx = j
+            # if we already rearranged vertexes we will not do it twice to prevent
+            # indexes mess
+            if min_d_idx != i and not ind_updated[i] and not ind_updated[min_d_idx]:
+                # swap vertexes
+                indexes[i] = min_d_idx
+                indexes[min_d_idx] = i
+                # mark vertexes as visited
+                ind_updated[i] = True
+                ind_updated[min_d_idx] = True
+
+        return indexes
+
+    def _filter_last_box(self, filter_speed):
+        if self.timestamps[-1] - self.timestamps[-2] == 1:
+            filtered_box = list(self.boxes[-2])
+            for j in range(len(self.boxes[-1])):
+                filtered_box[j] = int((1 - filter_speed) * filtered_box[j]
+                                      + filter_speed * self.boxes[-1][j])
+            self.boxes[-1] = tuple(filtered_box)
+
+    def add_detection(self, bbox, kps, timestamp, max_skip_size=1,
+                      box_filter_speed=0.7, kp_filter_speed=0.3,
+                      add_treshold=0.1, no_updated_frames_treshold=5):
+        skip_size = timestamp - self.get_end_time()
+        if 1 < skip_size <= max_skip_size:
+            self._interpolate(bbox, kps, timestamp, skip_size)
+            assert self.get_end_time() == timestamp - 1
+
+        self.boxes.append(bbox)
+        self.kps.append(kps)
+        self.timestamps.append(timestamp)
+        self._filter_last_box(box_filter_speed)
+        self._filter_last_3d_box(kp_filter_speed, add_treshold, no_updated_frames_treshold)
+
+
+class IOUTracker:
+    def __init__(self, sct_params):
+        self.time = 0
```

Comment:
May be I was unclear in the previous discussions, but we don't need this wrapper anymore and can actually turn `SingleCameraTracker` into `IOUTracker`

Reply:
done